File: truyenvkl/src/hot.lua
local url, page = ...

if text:is_empty(page) then
    page = 1
end
local doc = http:get(url .. "/page/" .. page):html()
if doc ~= nil then
    local el = doc:select(".theloai-thumlist li")
    local novelList = {}
    local next = num:to_int(doc:select(".pagination li.active + li"):text(), 1)
    local current = num:to_int(page, 1)

    if current >= next then
        next = ""
    end
    for i = 1, el:size() do
        local e = el:get(i - 1)
        local novel = {}
        local cover = e:select(".thumbnail img"):attr("data-src")
        if text:is_empty(page) then
            cover = e:select(".thumbnail img"):attr("src")
        end
        novel["name"] = e:select("h2"):text()
        novel["link"] = e:select("h2 a"):attr("href")
        novel["description"] = e:select(".content p"):first():text()
        novel["cover"] = cover
        novel["host"] = "https://truyenvkl.com"
        table.insert(novelList, novel)
    end

    return response:success(novelList, next)
end

return nil

File: truyenvkl/src/hot.js
function execute(url, page) {
    var doc;
    var urls = url.split(' ');
    if (!page) {
        page = '1';
        doc = Http.get(urls[0]).html().select(".theloai-thumlist");
    } else {
        var slug = url.match(/keyword\/(.*?)$/);
        doc = Http.post("https://s3.truyenhd.com/wp-admin/admin-ajax.php")
            .params({
                'action': "load_more_page_keyword",
                'current_page_tax': page,
                'the_loai': urls[1],
                'option_keyword_tax': urls[2]
            }).html();
    }

    if (doc) {
        var el = doc.select("li");
        var novelList = [];
        var next = parseInt(page) + 1;
        for (var i = 0; i< el.size(); i++) {
            var e = el.get(i);
            var cover = e.select(".thumbnail img").attr("data-src");
            if (!cover)
                cover = e.select(".thumbnail img").attr("src");
            novelList.push({
                name: e.select("h2").last().text(),
                link: e.select(" a").attr("href"),
                cover: cover,
                description: e.select(".content p").first().text(),
                host: "https://s3.truyenhd.com"
            });
        }

        return Response.success(novelList, next);
    }
    return null;
}